diffrax/_solver/srk.py

```diff
@@ -0,0 +1,624 @@
+    Should be a NumPy array of shape `(k,)`. Only used when `additive_noise=True`.
+- `bW`: Coefficient for the Brownian increment when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `bH`: Coefficient for the space-time Lévy area when computing the
+    output $y_{n+1}$. Should be a `Scalar`.
+- `aW`: The coefficients in front of the Brownian increment at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `aH`: The coefficients in front of the space-time Lévy area at each stage.
+    Should be of the same shape as `a`. Only used when `additive_noise=False`.
+- `additive_noise`: Whether the SDE has additive noise. If `True`, then `bW` and `bH`
+    should be scalars, and `cW` and `cH` should be arrays of shape `(k,)`. If `False`,
+    then `bW` and `bH` should be arrays of shape `(k,)`, `cW` and `cH` should be `None`,
+    and `aW` and `aH` should be lists of arrays of the same shape as `a`.
+"""
+
+
+class AbstractSRK(AbstractSolver[_SolverState]):
+    r"""A general Stochastic Runge-Kutta method.
+
+    The second term in the MultiTerm must be a `ControlTerm` with
+    `control=VirtualBrownianTree`. Depending on the Butcher tableau, the
+    `VirtualBrownianTree` may need to be initialised with 'levy_area="space-time"'
+    or 'levy_area="space-time-time"'.
+
+    Given the Stratonovich SDE
+    $dX_t = f(t, X_t) dt + g(t, X_t) \circ dW_t$
+
+    We construct the SRK with $s$ stages as follows:
+
+    $y_{n+1} = y_n + h \Big(\sum_{j=1}^s b_j f_j \Big)
+    + W_n \Big(\sum_{i=1}^{j-1} b^W_j g_i \Big)
+    + H_n \Big(\sum_{i=1}^{j-1} b^H_j g_i \Big)$
+
+    $f_j = f(t_0 + c_j h , z_j)$
+
+    $g_j = g(t_0 + c_j h , z_j)$
+
+    $z_j = y_n + h \Big(\sum_{i=1}^{j-1} a_{j,i} f_i \Big)
+    + W_n \Big(\sum_{i=1}^{j-1} a^W_{j,i} g_i \Big)
+    + H_n \Big(\sum_{i=1}^{j-1} a^H_{j,i} g_i \Big)$
+
+    where $W_n = W_{t_n, t_{n+1}}$ is the increment of the Brownian motion and
+    $H_n = H_{t_n, t_{n+1}}$ is its corresponding space-time Lévy Area.
+    A similar term can also be added for the space-time-time Lévy area, K.
+
+    In the special case, when the SDE has additive noise, i.e. when g is
+    independent of y (but can still depend on t), then the SDE can be written as
+    $dX_t = f(t, X_t) dt + g(t) \, dW_t$, and we can simplify the above to
+
+    $y_{n+1} = y_n + h \Big(\sum_{j=1}^s b_j k_j \Big) + g(t_n) \, (b^W
+    \, W_n + b^H \, H_n)$
+
+    $f_j = f(t_n + c_j h , z_j)$
+
+    $z_j = y_n + h \Big(\sum_{i=1}^{j-1} a_{j,i} f_i \Big) + g(t_n)
+    \, (c^W_j W_n + c^H_j H_n)$
+
+    When g depends on t, we need to add a correction term to $y_{n+1}$ of
+    the form $(g(t_{n+1}) - g(t_n)) \, (1/2 W_n - H_n)$.
+
+    The coefficients are provided in the `StochasticButcherTableau`.
+    """
+
+    term_structure: ClassVar = MultiTerm[tuple[ODETerm, AbstractTerm]]
+    interpolation_cls = LocalLinearInterpolation
```

Comment:
Do we really want to force this choice of `interpolation_cls` on *all* SRKs?

I think I'd suggest that this be a thing we require every subclass to declare. (This is also the convention used for the other solvers.)

Incidentally, it's a pity how many solvers are created without an eye to coming up with a dense interpolation scheme, which really is a necessary part of creating a full solver! Maybe nag James to add these to his solvers... ;)

Reply:
I'll discuss this with James tomorrow.

Reply:
We concluded that linear interpolation is probably the best choice at this point. Is there a way to interpolate between the stages as well?